tests/e2e/scenarios/test_reborn_webui_v2_tool_gates.py

```
@@ -0,0 +1,325 @@
+            reborn_v2_yolo_server,
+            thread_id,
+            timeout=60,
+        )
+        timeline = await fetch_timeline(client, reborn_v2_yolo_server, thread_id)
+
+    references = _tool_result_references(timeline)
+    assert references, timeline
+    assert any(reference.get("tool_result_ref") for reference in references), references
+    assert assistant.get("status") == "finalized", assistant
+    assistant_content = assistant.get("content")
+    assert isinstance(assistant_content, str) and assistant_content.strip(), assistant
```

Ruff (0.15.21)

[warning] Assertion should be broken down into multiple parts

(PT018)